src/qibocal/protocols/xyz_timing.py

```diff
@@ -0,0 +1,255 @@
+from dataclasses import dataclass, field
+
+import numpy as np
+import numpy.typing as npt
+import plotly.graph_objects as go
+from qibo.models.error_mitigation import curve_fit
+from qibolab import AcquisitionType, AveragingMode, Custom, Delay, Pulse, PulseSequence
+from scipy import special
+
+from qibocal.auto.operation import Data, Parameters, Results, Routine
+from qibocal.calibration.calibration import QubitId
+from qibocal.calibration.platform import CalibrationPlatform
+from qibocal.result import probability
+
+from .utils import table_dict, table_html
+
+COLORBAND = "rgba(0,100,80,0.2)"
+COLORBAND_LINE = "rgba(255,255,255,0)"
+PADDING_FLUX = 10
+
+
+@dataclass
+class XYZTimingResults(Results):
+    """Outputs of the xyz-timing protocol."""
+
+    fitted_parameters: dict[QubitId, list[float]]
+    """Output parameters of the fitted function."""
+    fitted_errors: dict[QubitId, list[float]]
+    """Errors of the fit parameters."""
+    delays: dict[QubitId, float]
+    """Flux-drive delays."""
+
+
+@dataclass
+class XYZTimingParameters(Parameters):
+    """XYZ-timing runcard inputs."""
+
+    flux_amplitude: float
+    """Amplitude of the flux pulse."""
+    delay_step: int
+    """Sweeper step value of the relative starts."""
+    delay_stop: int
+    """Sweeper stop value of the relative starts."""
+
+
+XYZTimingType = np.dtype(
+    [("delay", np.float64), ("prob", np.float64), ("errors", np.float64)]
+)
+
+
+@dataclass
+class XYZTimingData(Data):
+
+    pulse_duration: dict[QubitId, int]
+    """Duration of the drive and flux pulse"""
+    data: dict[QubitId, npt.NDArray] = field(default_factory=dict)
+    """Raw data acquired."""
+
+
+def _acquisition(
+    params: XYZTimingParameters,
+    platform: CalibrationPlatform,
+    targets: list[QubitId],
+) -> XYZTimingData:
+    """Data acquisition for drive-flux channels timing"""
+    natives = platform.natives.single_qubit
+    durations = np.arange(
+        1,
+        params.delay_stop,
+        params.delay_step,
+    )
+    sequences = []
+    ro_pulses = []
+    drive_durations = {}
+    for qubit in targets:
+        drive_pulse = natives[qubit].RX()[0]
+        drive_durations[qubit] = int(drive_pulse[1].duration)
+
+    data = XYZTimingData(pulse_duration=drive_durations)
+    for duration in durations:
+        ro_pulses.append([])
+        for qubit in targets:
+            sequence = PulseSequence()
+            drive_channel = platform.qubits[qubit].drive
+            flux_channel = platform.qubits[qubit].flux
+            ro_channel = platform.qubits[qubit].acquisition
+            drive_pulse = natives[qubit].RX()[0]
+            readout_pulse = natives[qubit].MZ()[0]
```

Comment:
Here I would specify that you get both the channel and the pulse.
```suggestion
            drive_channel, drive_pulse = natives[qubit].RX()[0]
            readout_channel, readout_pulse = natives[qubit].MZ()[0]
```
In this way you don't need to access the acquisition and the drive channel directly as you are doing a few lines above.